### DISCUSSION TITLE
opt.formatoptions:remove({ "c", "r", "o" })

### DISCUSSION THREAD

**Discussion by yjchen101** [LazyVim/LazyVim#3018] [Q&A] [answered]
I configured opt.formatoptions:remove({ "c", "r", "o" }) in options.lua, but it didn't actually take effect
<img width="902" alt="image" src="https://github.com/LazyVim/LazyVim/assets/75558384/467ccf4f-8c7f-4379-a9f2-9f1e9e4c2e21">

**Comment by dpetka2001** [comment, marked as answer]
You have to put that in an autocommand on `Filetype` to take effect.

**Reply by yjchen101** [reply]
I succeeded with your method, thank you🙏

**Reply by dpetka2001** [reply]
In case you're wondering why this happens, it's because of the Neovim [runtime ftplugins](https://github.com/neovim/neovim/tree/master/runtime/ftplugin). You can see for example for [Lua](https://github.com/neovim/neovim/blob/9319d97679053a219b1701d70abce414d0f6a8b0/runtime/ftplugin/lua.vim#L20) that it sets `formatoptions` of its own. Other ftplugins do the same. That's why the autocmd is needed in this specific case.
